citation_helper_service/citation_helper.py

```diff
@@ -36,19 +37,19 @@ def get_suggestions(**args):
     if "Error" in papers:
         return papers
     # removes papers from the original list to get candidates
-    papers = filter(lambda a: a not in bibcodes, papers)
+    papers = [a for a in papers if a not in bibcodes]
     # establish frequencies of papers in results
     paperFreq = [(k, len(list(g))) for k, g in groupby(sorted(papers))]
     # and sort them, most frequent first
     paperFreq = sorted(paperFreq, key=operator.itemgetter(1), reverse=True)
     # remove all papers with frequencies smaller than threshold
-    paperFreq = filter(lambda a: a[1] > current_app.config.get(
-        'CITATION_HELPER_THRESHOLD_FREQUENCY'), paperFreq)
+    paperFreq = [a for a in paperFreq if a[1] > current_app.config.get(
+        'CITATION_HELPER_THRESHOLD_FREQUENCY')]
     # get metadata for suggestions
     meta_dict = get_meta_data(results=paperFreq[:Nsuggestions])
     if "Error"in meta_dict:
         return meta_dict
     # return results in required format
     return [{'bibcode': x, 'score': y, 'title': meta_dict[x]['title'],
              'author':meta_dict[x]['author']} for (x, y) in
-            paperFreq[:Nsuggestions] if x in meta_dict.keys()]
+            paperFreq[:Nsuggestions] if x in list(meta_dict.keys())]
```

Comment:
I think this `list(...)` (automatically added by the porting tools) might not be needed here.